src/vec.rs

```diff
@@ -28,20 +34,41 @@ use core::{cmp::Ordering, fmt, hash, iter::FromIterator, mem::MaybeUninit, ops,
 /// }
 /// assert_eq!(*vec, [7, 1, 2, 3]);
 /// ```
-pub struct Vec<T, const N: usize> {
+pub struct Vec<T, const N: usize, A = u8> {
     // NOTE order is important for optimizations. the `len` first layout lets the compiler optimize
     // `new` to: reserve stack space and zero the first word. With the fields in the reverse order
     // the compiler optimizes `new` to `memclr`-ing the *entire* stack space, including the `buffer`
     // field which should be left uninitialized. Optimizations were last checked with Rust 1.60
     len: usize,
 
-    buffer: [MaybeUninit<T>; N],
+    storage: VecBuf<T, N, A>,
```

Comment:
could you keep the name `buffer` to keep the diff small?

Reply:
@Dirbaio Fixed.